src/bin/sol-fbp-generator/main.c

```diff
@@ -802,35 +820,38 @@ generate(struct sol_vector *fbp_data_vector)
     out(
         "static void\n"
         "initialize_types(void)\n"
-        "{\n");
-    SOL_VECTOR_FOREACH_IDX (&ctx->types_to_initialize, symbol, i) {
+        "{\n"
+        "    const struct sol_flow_node_type **t;\n\n");
+    SOL_VECTOR_FOREACH_IDX (&ctx->types_to_initialize, type, i) {
         out(
-            "    if (%.*s->init_type)\n"
-            "        %.*s->init_type();\n",
-            SOL_STR_SLICE_PRINT(*symbol),
-            SOL_STR_SLICE_PRINT(*symbol));
+            "    t = sol_symbol_get(\"flow\", \"%.*s\", %.*s);\n"
```

Comment:
On Fri, 18 Sep 2015 14:47:44 -0700, Gustavo Sverzut Barbieri wrote:
> >          out(
> > -            "    if (%.*s->init_type)\n"
> > -            "        %.*s->init_type();\n",
> > -            SOL_STR_SLICE_PRINT(*symbol),
> > -            SOL_STR_SLICE_PRINT(*symbol));
> > +            "    t = sol_symbol_get(\"flow\", \"%.*s\", %.*s);\n"
> 
> `if (t)` to check missing symbols?

Yes, but I ended up wondering if that whole thing shouldn't be
re-adapted to consider that symbols might be missing, and do the
symbol_get thing only in one place. I'll look into that after.